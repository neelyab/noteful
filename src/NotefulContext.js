import React from 'react'

const NotefulContext = React.createContext({
    folders:[],
    notes:[],
    addFolder: ()=>{},
    deleteNote: ()=>{},
    deleteFolder: ()=>{},
})

export default NotefulContext;