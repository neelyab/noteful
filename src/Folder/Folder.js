import React, {Component} from 'react';
import './Folder.css'
import {NavLink} from 'react-router-dom';
import NotefulContext from '../NotefulContext'

function handleDeleteFolder(id, cb){
    console.log(id);
    console.log(cb)
    fetch( `http://localhost:9090/folders/${id}`, {
        method: 'DELETE',
        headers: {
            'content-type': 'application/json'
        },
    })
    .then (res => {
        if (!res.ok){
            return res.json().then(error=> {
                throw error
            })
        }
        return res.json()
    })
    .then(data=> {
        console.log(data);
        cb(id);

    })
    .catch(error=> {
        console.log(error)
    })
}

class Folder extends Component {
    static contextType = NotefulContext;
    render(){
        const folderToHighlight = this.context.folders.find(folder=>folder.id===this.props.routerProps.folderId) || {};
        return (
            <NotefulContext.Consumer>
                {(context)=> {
                return (
            <li key={this.props.folderId}
            className={this.props.folderId === folderToHighlight.id ? 'folder highlighted' : 'folder'}>
                <NavLink to={this.props.link}>
                {this.props.name}
                </NavLink>
                {/* <button className="delete" id={this.props.id} onClick={()=>handleDeleteFolder(this.props.folderId, context.deleteFolder)}>Delete</button> */}
                </li>
                )
                }
            }
                </NotefulContext.Consumer>
        )
    }
}
export default Folder;