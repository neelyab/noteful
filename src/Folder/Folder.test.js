import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'
import Folder from './Folder'
import NotefulContext from '../NotefulContext';

const routerp =  {"folderId": "b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1",
"name": "Important"}

const folders = [
    {
      "id": "b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1",
      "name": "Important"
    },
    {
      "id": "b07161a6-ffaf-11e8-8eb2-f2801f1b9fd1",
      "name": "Super"
    },
    {
      "id": "b07162f0-ffaf-11e8-8eb2-f2801f1b9fd1",
      "name": "Spangley"
    }
  ];

it.skip('renders to UI as expected', ()=> {

const tree = renderer
.create(
<NotefulContext.Provider value={folders}>
    <Folder routerProps={routerp} folderId="b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1" />
</NotefulContext.Provider>)
.toJSON()
expect(tree).toMatchSnapshot();
})