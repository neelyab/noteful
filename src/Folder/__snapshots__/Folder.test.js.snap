// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`folder component renders to UI as expected 1`] = `
<li
  className="folder highlighted"
>
  <a
    aria-current={null}
    href="/link"
    onClick={[Function]}
  />
</li>
`;
